src/hooks/connectWeb3/useWeb3React.tsx

```diff
@@ -104,12 +104,22 @@ export function ActiveWeb3Provider({
     [jsonRpcUrlMap, defaultChainId]
   )
 
-  connections = [metaMaskConnection, walletConnectConnectionQR, walletConnectConnectionPopup, networkConnection]
-  if (integratorConnection) connections = [integratorConnection, ...connections]
+  const key = useRef(0)
+  connections = useMemo(() => {
+    key.current += 1
```

Comment:
this feels a little funny, i'd be happy to put this in a useEffect as you suggested @zzmp, but incrementing the key every time this memo function runs should accomplish what we're going for, and it felt more tightly couple to the memoization

Reply:
TBH I think this is most cleanly expressed by splitting the memo and the hook, because that's the most "react"-ey and best expresses the intentions:

    const connections = useMemo(() =>
      [integratorConnection, metaMaskConnection, walletConnectConnectionQR, walletConnectConnectionPopup, networkConnection].filter(Boolean),
      [integratorConnection, metaMaskConnection, walletConnectConnectionQR, walletConnectConnectionPopup, networkConnection])
    useEffect(() => key.current += 1, [connections])

Reply:
also I'm very on the fence of splitting the useMemo into useMemo/useEffect, so I think you should probably ignore my last comment and leave them coupled. coupling here feels good.

Reply:
going to leave in the useMemo for now, added an explanatory comment!